cmd/object-handlers.go

```diff
@@ -689,6 +689,167 @@ func (api objectAPIHandlers) getObjectHandler(ctx context.Context, objectAPI Obj
 	})
 }
 
+// GetObjectAttributes ...
+func (api objectAPIHandlers) getObjectAttributesHandler(ctx context.Context, objectAPI ObjectLayer, bucket, object string, w http.ResponseWriter, r *http.Request) {
+	opts, valid := getAndValidateAttributesOpts(ctx, w, r, bucket, object)
+	if !valid {
+		return
+	}
+
+	var s3Error APIErrorCode
+	if opts.VersionID != "" {
+		s3Error = checkRequestAuthType(ctx, r, policy.GetObjectVersionAttributesAction, bucket, object)
+		if s3Error == ErrNone {
+			s3Error = checkRequestAuthType(ctx, r, policy.GetObjectVersionAction, bucket, object)
+		}
+	} else {
+		s3Error = checkRequestAuthType(ctx, r, policy.GetObjectAttributesAction, bucket, object)
+		if s3Error == ErrNone {
+			s3Error = checkRequestAuthType(ctx, r, policy.GetObjectAction, bucket, object)
+		}
+	}
+
+	if s3Error != ErrNone {
+		writeErrorResponse(ctx, w, errorCodes.ToAPIErr(s3Error), r.URL)
+		return
+	}
+
+	objInfo, err := objectAPI.GetObjectInfo(ctx, bucket, object, opts)
+	if err != nil {
+
+		reqInfo := logger.GetReqInfo(ctx)
+		if reqInfo == nil {
+			writeErrorResponseHeadersOnly(w, errorCodes.ToAPIErr(ErrAccessDenied))
+			return
+		}
+
+		if globalIAMSys.IsAllowed(policy.Args{
+			AccountName:     reqInfo.Cred.AccessKey,
+			Groups:          reqInfo.Cred.Groups,
+			Action:          policy.ListBucketAction,
+			BucketName:      bucket,
+			ConditionValues: getConditionValues(r, "", reqInfo.Cred),
+			ObjectName:      object,
+			IsOwner:         reqInfo.Owner,
+			Claims:          reqInfo.Cred.Claims,
+		}) {
+			writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL)
+			return
+		}
+		writeErrorResponse(ctx, w, errorCodes.ToAPIErr(ErrAccessDenied), r.URL)
```

Comment:
Do we really need this code? I mean wouldn't all permission errors returned by checkRequestAuthType() already ? (or add a comment)

Reply:
you are right, not sure why I didn't opt to use the checkRequestAuthType for that check as well. Will fix.